fastjet/clustersequence.go

```diff
@@ -360,11 +370,30 @@ func (cs *ClusterSequence) runN3Dumb() error {
 					cs.jetScaleForAlgorithm(ijet),
 					cs.jetScaleForAlgorithm(jjet),
 				)
-				y := jetscale * Distance(ijet, jjet) * cs.invR2
-				if y < ymin {
-					ymin = y
-					ii = i
-					jj = j
+				switch cs.alg {
+				case EeGenKtAlgorithm:
+					den := 1 - math.Cos(cs.r)
+					if cs.r > math.Pi {
+						den = 3 + math.Cos(cs.r)
+					}
+					y := math.MaxFloat64
+					if den != 0 {
+						y = jetscale * (1 - fmom.CosTheta(&ijet.PxPyPzE, &jjet.PxPyPzE)) / den
+					}
+					if y < ymin {
+						ymin = y
+						ii = i
+						jj = j
+					}
+					break
```

Comment:
Unlike `C`, the various `case` blocks of a `switch` control structure do not need any `break` in `Go`.
see:
 https://golang.org/ref/spec#Switch_statements

```
In an expression switch, the switch expression is evaluated and the case expressions, which need not be constants, are evaluated left-to-right and top-to-bottom; the first one that equals the switch expression triggers execution of the statements of the associated case; the other cases are skipped.
```

just remove that `break`